cloud-controller-manager/cmd/digitalocean-webhook-server/main.go

```diff
@@ -40,25 +102,40 @@ func main() {
 }
 
 func startWebhookServer() error {
-	// default server running at port 9443, looking for tls.crt, tls.key in /tmp/k8s-webhook-server/serving-certs
-	ll := ctrl.Log.WithName("webhook-validation-server")
-	ll.Info("getting config")
-	config, err := ctrl.GetConfig()
-	if err != nil {
-		fmt.Fprintf(os.Stderr, "failed to retrieve kubeconfig %v\n", err)
-	}
+
+	ll := zap.New(logOpts...).WithName("webhook-validation-server")
+	ctrlruntimelog.SetLogger(ll)
+
 	server := webhook.Server{}
-	c, err := client.New(config, client.Options{Scheme: scheme})
+
+	doClient, err := initDOClient()
 	if err != nil {
-		fmt.Fprintf(os.Stderr, "failed to construct cr client %v\n", err)
+		ll.Error(err, "failed to initialize DO client")
+		os.Exit(1)
 	}
 
-	server.Register("/validate-doks-lb-service", &webhook.Admission{Handler: &do.DOKSLBServiceValidator{Client: c, Log: ll}})
-	ll.Info("Registering Webhook server handlers")
-	if err = server.StartStandalone(ctrl.SetupSignalHandler(), scheme); err != nil {
-		return err
+	clusterID := os.Getenv(doClusterIDEnv)
```

Comment:
It seems like we're using this for the tag of the LB's target droplets. At some point, I suggest to not conflate the cluster ID parameter with the droplet tag: this happens to be same for DOKS clusters but certainly does not have to be for DIY clusters. So my suggestion would be to have a new, dedicated environment flag for the webhook (like `DROPLET_TAG`) and use that.

I'm still not overly fond of the idea to use a tag and would much rather get rid of the need to having to specify a droplet at all; but let's defer that discussion to later. My prime interest here is that we do not couple existing parameters we later need to uncouple again.